File: Tree/InvertBinaryTree.py
class Node:
	def __init__(self, val):
		self.val = val
		self.left = None
		self.right = None

class Binary_Tree:
	def __init__(self):
		self.root = None
		self.size = 0

def invert_BT(node):
	if not node:
		return
	node.left, node.right = node.right, node.left
	invert_BT(node.left)
	invert_BT(node.right)

bt = Binary_Tree()
a = Node('a')
b = Node('b')
c = Node('c')
d = Node('d')
e = Node('e')

a.left = b
a.right = c
c.left = d
c.right = e

print(a.val)
print(a.left.val)
print(a.right.val)
print(a.right.left.val)
print(a.right.right.val)

invert_BT(a)

print(a.val)
print(a.left.val)
print(a.right.val)
print(a.left.left.val)
print(a.left.right.val)

File: Tree/InvertBinaryTree.js
function Node(value){
	this.value = value;
	this.left = null;
	this.right = null;
}

function invertBT(node){
	if(node !== null){
		var temp = node.left;
		node.left = node.right;
		node.right = temp;
		invertBT(node.left);
		invertBT(node.right);
	}
}

var a = new Node(10);
var b = new Node(5);
var c = new Node(15);
var d = new Node(3);
var e = new Node(7);
var f = new Node(13);
var g = new Node(17);

a.left = b;
a.right = c;
b.left = d;
b.right = e;
c.left = f;
c.right = g;

function traverseBT(node){
	if(node !== null){
		console.log(node.value);
		traverseBT(node.left);
		traverseBT(node.right);
	}
}

traverseBT(a);
invertBT(a);
traverseBT(a);